### PR TITLE
Switched to promise.

### PR DESCRIPTION
**Pull request by simonmika**

### CODE REVIEW

File: src/components.d.ts [hunk 1]
```diff
@@ -23,7 +26,7 @@ export namespace Components {
 
   interface Card3Form {
     'state': "failed" | "succeeded" | "processing" | "created";
-    'submit': (method: string, amount: number, currency: Currency, order?: string | undefined, reference?: string | undefined, description?: string | undefined) => void;
+    'submit': (method: import("/home/smika/versioned/github/certitrade/card3-form/node_modules/@certitrade/card3-model/dist/Method").Method, amount: number, currency: Currency, order?: string | undefined, reference?: string | undefined, description?: string | undefined) => Promise<Payment>;
```
Comment (johnnysolsson) [inline src/components.d.ts:29]: Hardcoded repo?

Comment (simonmika) [reply src/components.d.ts:29]: Oops, good catch.

Comment (simonmika) [reply src/components.d.ts:29]: It's in the autogenerated `Components.d.ts` file. Maybe time to try to remove it from the repo?

Comment (johnnysolsson) [reply src/components.d.ts:29]: Yes. Maybe.

**Review by johnnysolsson — APPROVED** [review]

**Pull request merged**